clang/lib/Parse/ParseDecl.cpp

```diff
@@ -1842,8 +1842,7 @@ void Parser::ProhibitCXX11Attributes(ParsedAttributes &Attrs,
       continue;
     if (AL.getKind() == ParsedAttr::UnknownAttribute) {
       if (WarnOnUnknownAttrs)
-        Diag(AL.getLoc(), diag::warn_unknown_attribute_ignored)
-            << AL << AL.getRange();
+        Actions.DiagnoseUnknownAttribute(AL);
```

Comment:
Curious that this is not setting it to invalid, but the branch below is.  Is that a mistake?